src/UserSpaceInstrumentation/Attach.cpp

```diff
@@ -111,7 +97,32 @@ ErrorMessageOr<void> AttachAndStopProcess(pid_t pid) {
     }
     process_tids = GetTidsOfProcess(pid);
   }
-  return outcome::success();
+  return halted_tids;
+}
+
+}  // namespace
+
+ErrorMessageOr<absl::flat_hash_set<pid_t>> AttachAndStopProcess(pid_t pid) {
+  ErrorMessageOr<pid_t> error_or_tracer_pid = orbit_base::GetTracerPidOfProcess(pid);
+  if (error_or_tracer_pid.has_error()) {
+    return ErrorMessage(absl::StrFormat("There is no process with pid %d: %s", pid,
+                                        error_or_tracer_pid.error().message()));
+  }
+
+  const pid_t tracer_pid = error_or_tracer_pid.value();
+  if (tracer_pid != 0) {
+    return ErrorMessage(
+        absl::StrFormat("Process %d is already being traced by %d. Please make sure no debugger is "
+                        "attached to the target process when profiling.",
+                        pid, tracer_pid));
+  }
+
+  return AttachAndStopThreads(pid, {});
+}
+
+[[nodiscard]] ErrorMessageOr<absl::flat_hash_set<pid_t>> AttachAndStopNewThreadsOfProcess(
```

Comment:
Why this forwarding? `AttachAndStopNewThreadsOfProcess` is identical to `AttachAndStopThreads` above.

You coukld rename `AttachAndStopThreads` to  `AttachAndStopNewThreadsOfProcess` and delete this.

Reply:
Oh right. On the one hand, since the two public functions would be used for slightly different purposes, it might be conceptually cleaner to have a common hidden utility function used by both. On the other hand I prefer the simplicity of your suggestion, so I changed it.